lib/open-graph.ts

```diff
@@ -0,0 +1,55 @@
+import { ObjectLiteral, OpenGraphData } from "@/types/index";
+import { isImageType, isPublicWork } from "./work-helpers";
+import { PRODUCTION_URL } from "@/lib/constants/endpoints";
+import { WorkShape } from "@/types/components/works";
+import { overviewThumbnails } from "@/lib/constants/homepage";
+
+export const defaultOpenGraphImage = overviewThumbnails[0][0].id;
+
+export const defaultOpenGraphData: OpenGraphData = {
+  "og:description":
+    "Enrich your research with primary sources. Explore millions of high-quality primary sources and images from around the world, including artworks, maps, photographs, and more.",
+  "og:image": defaultOpenGraphImage,
+  "og:image:secure_url": defaultOpenGraphImage,
+  "og:site_name": "Digital Collections - Libraries - Northwestern University",
+  "og:title": "Digital Collections - Libraries - Northwestern University",
+  "og:type": "website",
+  "og:url": PRODUCTION_URL,
+};
+
+export function buildWorkOpenGraphData(
+  work: WorkShape | null
+): ObjectLiteral | OpenGraphData {
+  if (!work) return {};
+
+  const ogTitle = work?.title ? work.title : work?.accession_number;
+  const isPublic = isPublicWork(work?.visibility);
+  const imageUrl =
+    isPublic && isImageType(work?.work_type)
+      ? `${work?.representative_file_set.url}/full/1200,630/0/default.jpg`
+      : work?.thumbnail;
+
+  let ogDescription = "";
+  if (work?.description && work.description.length > 0) {
+    ogDescription = `${work.description[0]} - `;
```

Comment:
Is it possible for us to want to deliver all values of a description? Maybe something like this in case?

```suggestion
    ogDescription = `${work.description.join(" ")} - `;
```

Reply:
Sure thing